### PR TITLE
Require hook maturity level on hook definition

### LINKED ISSUE #444: Need to add hook maturity to the hook definition and template
Now that the hook maturity model is part of the specification, hook definitions need to include the maturity level for each hook as part of the hook metadata, and the hook definition template needs to be updated to include the default hook maturity, `0 - draft`.

### PR DESCRIPTION
**Pull request by isaacvetter**
Fixes #444

### CODE REVIEW

**Comment by isaacvetter** [comment]
I specifically left maturity changes out of the required change log, because optimally a hook should mature without substantive changes.

File: docs/specification/1.0.md [hunk 1]
```diff
@@ -873,7 +873,9 @@ Hook creators SHOULD include examples of the context.
...
+The intent of the CDS Hooks Maturity Model is to attain broad community engagement and consensus, before a hook is labeled as mature, that the hook is necessary, implementable, and worthwhile to the CDS services and CDS clients that would reasonably be expected to use it. Implementer feedback should drive the maturity of new hooks. Diverse participation in open developer forums and events, such as HL7 FHIR Connectathons, is necessary to achieve significant implementer feedback. The below criteria will be evaluated with these goals in mind. The hook definition defines the hook's current maturity level and MUST link to the definition of that maturity level.
```
Comment (kpshek) [inline docs/specification/1.0.md]: > and MUST link to the definition of that maturity level.

I thought the important thing is that a hook definition MUST define its hook maturity level.

I'm not sure we should require the maturity level to be linked. Yes, linking is nice but we don't require it for anything else (like say what "prefetch token" means). I feel like that for the concept of Maturity Model, along with every other concept we have defined in CDS Hooks, requires the reader of the hook to refer back to the spec to understand everything completely.

**Comment by isaacvetter** [comment]
Hey @kpshek - can you take another look at this PR? (and then #451) ?

**Comment by kpshek** [comment]
Thank you @isaacvetter and sorry for re-reviewing this so late!

**Pull request merged**